### DISCUSSION TITLE
Can I load the maps from the file storage instead of downloading them?

### DISCUSSION THREAD

**Discussion by tom-programming** [organicmaps/organicmaps#8025] [Q&A] [answered]
I am trying to have an airgapped Android device. I wonder if I can upload the maps that OM needs via an USB cable or via Bluettoth without requiring internet access from the device, in a simple way.

If this is impossible now but it can be done with some tinkering with the source code I'd be thankful for any guidance too

I guess something can be done by mirroring an http server https://cdn.organicmaps.app but the app opens directly with the "Download the world map" page for me which does not allow me to access any settings where I could direct it to localhost or similar

**Comment by biodranik** [comment]
Did you try to choose the external storage path in OM storage settings, and then use `adb push` to upload data there via USB cable?

**Reply by tom-programming** [reply]
I tried but I don't think I have access to the settings. It's just a page prompting me to download the world map

**Reply by biodranik** [reply, marked as answer]
Try to install this version first, it has an embedded world map, to see the actual storage path (and let us know here too if you managed to upload maps, including world files, using adb push):
https://cdn.organicmaps.app/apk/Organic-Maps-Google-24033108.apk

**Reply by tom-programming** [reply]
Thanks! Using this apk I was able to get to the next stage, alas it still didn't work

I tried to load a mwm file in the shared storage, having looked at the structure I have in a different device,  but it failed

- First I verified that the setting for the map storage is set to shared storage
- If I try to download a map of PLACE first from the app it creates a YYMMDD directory in /storage/emulated/0/Android/data/app.organicmaps/files/ and inside it there are PLACE.mwm.ready.downloading and PLACE.mwm.ready.resume files. I cannot delete them and I cannot upload there the PLACE.mwm neither with `adb push` nor with Android File Transfer. The directory permissions are `drwxr-s---` so it is logical I cannot add anything to it
- If instead I try to create the YYMMDD  directory myself before the app creates it I can upload there  PLACE.mwm but still the app ignores it and instead prompts me to download that particular map piece. As before it creates PLACE.mwm.ready.downloading and PLACE.mwm.ready.resume files completely ignoring the PLACE.mwm that is there

Presumably I can root the device but I'd be happy if there was a way without doing that

**Reply by biodranik** [reply]
Make sure that the map data version supported by app matches uploaded mwm files.

Did you restart the app after uploading file?

**Reply by tom-programming** [reply]
I didn't (erm...)

Now I did it and it works, it loaded the maps and I can navigate in them, thanks!

Still it is dependent on that preloaded apk you shared but I solved my problem

**Comment by hb0nd** [comment]
If it's air gapped, you might want to consider getting root via magisk, then you can easily transfer map files 👍
